mteb/evaluation/evaluators/RerankingEvaluator.py

```diff
@@ -179,39 +189,82 @@ def compute_metrics_individual(self, model):
+            scores = self._compute_metrics_instance(sim_scores, is_relevant)
+            conf_scores = confidence_scores(sim_scores)
+
             all_mrr_scores.append(scores["mrr"])
             all_ap_scores.append(scores["ap"])
+            all_conf_scores.append(conf_scores)
 
         mean_ap = np.mean(all_ap_scores)
         mean_mrr = np.mean(all_mrr_scores)
 
-        return {"map": mean_ap, "mrr": mean_mrr}
+        # Compute nAUCs
+        naucs_map = self._compute_nAUCs(all_conf_scores, all_ap_scores, "map")
+        naucs_mrr = self._compute_nAUCs(all_conf_scores, all_mrr_scores, "mrr")
 
-    def _compute_metrics_instance(self, query_emb, docs_emb, is_relevant):
-        """Computes metrics for a single instance = (query, positives, negatives)
+        return {**{"map": mean_ap, "mrr": mean_mrr}, **naucs_map, **naucs_mrr}
+
+    def _compute_sim_scores_instance(self, query_emb, docs_emb):
+        """Computes similarity scores for a single instance = (query, positives, negatives)
 
         Args:
             query_emb (`torch.Tensor` of shape `(num_queries, hidden_size)`): Query embedding
                 if `num_queries` > 0: we take the closest document to any of the queries
             docs_emb (`torch.Tensor` of shape `(num_pos+num_neg, hidden_size)`): Candidates documents embeddings
+
+        Returns:
+            sim_scores (`torch.Tensor of shape `(num_pos+num_neg,)`): Query-documents similarity scores
+        """
+        sim_scores = self.similarity_fct(query_emb, docs_emb)
+        if len(sim_scores.shape) > 1:
+            sim_scores = torch.amax(sim_scores, dim=0)
+
+        return sim_scores
+
+    def _compute_metrics_instance(self, sim_scores, is_relevant):
+        """Computes metrics for a single instance = (query, positives, negatives)
+
+        Args:
+            sim_scores (`torch.Tensor of shape `(num_pos+num_neg,)`): Query-documents similarity scores
             is_relevant (`List[bool]` of length `num_pos+num_neg`): True if the document is relevant
 
         Returns:
             scores (`Dict[str, float]`):
                 - `mrr`: Mean Reciprocal Rank @ `self.mrr_at_k`
                 - `ap`: Average Precision
         """
-        pred_scores = self.similarity_fct(query_emb, docs_emb)
-        if len(pred_scores.shape) > 1:
-            pred_scores = torch.amax(pred_scores, dim=0)
-
-        pred_scores_argsort = torch.argsort(-pred_scores)  # Sort in decreasing order
-
+        pred_scores_argsort = torch.argsort(-sim_scores)  # Sort in decreasing order
         mrr = self.mrr_at_k_score(is_relevant, pred_scores_argsort, self.mrr_at_k)
-        ap = self.ap_score(is_relevant, pred_scores.cpu().tolist())
+        ap = self.ap_score(is_relevant, sim_scores.cpu().tolist())
         return {"mrr": mrr, "ap": ap}
 
+    def _compute_nAUCs(self, all_conf_scores, metrics, metric_name):
+        """Computes nAUCs given confidence scores and evaluation metrics all samples
```

Comment:
It would be lovely with a bit more explanation and potentially a ref.

Reply:
I have provided some explanations (+ a reference to the relevant paper) on nAUC in mteb/evaluation/evaluator/utils.py, were the function is originally defined.